### ISSUE TITLE
reload matchengine config file

### ISSUE THREAD

**Issue by TAiS46** [viabtc/viabtc_exchange_server#92]
after adding a new currency / market, how can I reload the config file without restarting the server?

I tried to send SIGUSR1 as to `https://github.com/viabtc/viabtc_exchange_server/blob/master/utils/ut_signal.c` but didn't work

**Comment by haipome** [comment]
It require restart in current version.

**Comment by yunhui0501** [comment]
我增加了一个接口，当收到要增加新的资产和币种，我会再原有的字典哈希表中增加。

**Comment by ohld** [comment]
Any updates on this? 
@TAiS46  @haipome What is the best way to restart the exchange to reload the config? Thanks.

**Comment by yunhui0501** [comment]
这个要看需求，我们做的是24小时撮合，所以不能用重启来解决。

At 2018-08-12 06:09:44, "Daniil Okhlopkov" <notifications@github.com> wrote:


Any updates on this?
@TAiS46@haipome What is the best way to restart the exchange to reload the config? Thanks.

**Comment by yunhui0501** [comment]
不是，比如IP,端口等等不经常改变的，用的还是配置文件。经常改变的，需要通信，在数据库和内存中同时修改。





At 2018-08-12 06:09:44, "Daniil Okhlopkov" <notifications@github.com> wrote:


Any updates on this?
@TAiS46@haipome What is the best way to restart the exchange to reload the config? Thanks.

**Comment by githubzb** [comment]
@yunhui0501  请问你是如何做到增加新的资产和币种的，可以分享一下给我吗

**Comment by objectt** [comment]
I also have a system where new assets/markets can be added dynamically. I am working on a fix that periodically reloads from database. https://github.com/objectt/viabtc_exchange_server

**Comment by ohld** [comment]
May be we can create a new version of viaBTC engine with all that user customisations? We need to implement some tests to cover at least a half of the functionality. And we can make this engine work just out of the box.

**Comment by objectt** [comment]
@ohId yes, i agree. We could make a better well-designed open source product together.

**Comment by ohld** [comment]
@objectt Contact me Okhlopkov.com

**Comment by yunhui0501** [comment]
@githubzb 增加一个接口，参数有币种和资产数量，一面在数据库增加，一面在内存里增加。

**Comment by ohld** [comment]
@objectt I have not received any message from you. If someone wants to unite the forces to build the huge update to viaBTC engine, contact me danokhlopkov (at) gmail.com or https://t.me/okhlopkov.

First goal can be to find and merge the best improvements and bug fixes from all forks.

**Comment by aishwarydhare** [comment]
@ohld I am up to contribute, Is there any existing active fork?

**Comment by ohld** [comment]
Actually, no. But my fork is the latest and freshest. Feel free to contribute to my fork. Thanks.

**Comment by ohld** [comment]
@aishwarydhare by my fork I means fork from @testnet-exchange
